index.html

```diff
@@ -83,95 +69,172 @@
     <div id="chart"></div>
   </div>
 </div>
-<script type="text/javascript" src="https://npmcdn.com/d3@3.5.17/d3.min.js"></script>
+<script type="text/javascript" src="https://npmcdn.com/d3@3.5.17/d4.min.js"></script>
 <script type="text/javascript" src="https://npmcdn.com/science@1.9.3/science.v1.min.js"></script>
 <script type="text/javascript" src="https://npmcdn.com/simple-statistics@2.0.0/dist/simple-statistics.min.js"></script>
+<script type="text/javascript" src="https://npmcdn.com/d3-tip@0.6.7/index.js"></script>
 <script>
   var data = [];
   var options = {
-    yearRange: 'all'
+    location: 'all'
   };
-
   var svg, x, y;
-
-  var margin = { top: 15, right: 15, bottom: 30, left: 45 };
-
-  var width = 640 - margin.right - margin.left;
-  var height = 400 - margin.top - margin.bottom;
-
+  var margin = { top: 15, right: 15, bottom: 30, left: 90 };
+  var width = 900 - margin.right - margin.left;
+  var height = 500 - margin.top - margin.bottom;
+  var binCount = 54
+  var binCounty = 15
   // FETCH DATA
-
   d3.json('data.json', function (error, json) {
     if (error) { throw error; }
-
     data = json;
-
-    setup();
-    update();
+    var chart = new Chart();
   });
```

Comment:
You also want to call chart.update() right after instantiating chart, see below for more on this.